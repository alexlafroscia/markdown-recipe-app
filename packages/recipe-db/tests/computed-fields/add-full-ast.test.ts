import { expect, test } from 'vitest';
import dedent from 'dedent';

import { makeClient, withRecipe } from '../helpers';

const WITH_INGREDIENT_HEADER = dedent`
    ## Ingredients

    - 1 small onion
    - 1/2 lb ground beef
`;

test('attaching the full AST', async () => {
	const client = await makeClient(withRecipe(WITH_INGREDIENT_HEADER));
	const [recipe] = await client.recipes();

	expect(recipe.fullAST).toMatchSnapshot();
});
