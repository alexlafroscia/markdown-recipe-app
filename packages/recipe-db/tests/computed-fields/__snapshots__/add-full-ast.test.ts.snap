// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`attaching the full AST 1`] = `
{
  "children": [
    {
      "children": [
        {
          "position": {
            "end": {
              "column": 15,
              "line": 1,
              "offset": 14,
            },
            "start": {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
          },
          "type": "text",
          "value": "Ingredients",
        },
      ],
      "depth": 2,
      "position": {
        "end": {
          "column": 15,
          "line": 1,
          "offset": 14,
        },
        "start": {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "heading",
    },
    {
      "children": [
        {
          "checked": null,
          "children": [
            {
              "children": [
                {
                  "position": {
                    "end": {
                      "column": 16,
                      "line": 3,
                      "offset": 31,
                    },
                    "start": {
                      "column": 3,
                      "line": 3,
                      "offset": 18,
                    },
                  },
                  "type": "text",
                  "value": "1 small onion",
                },
              ],
              "position": {
                "end": {
                  "column": 16,
                  "line": 3,
                  "offset": 31,
                },
                "start": {
                  "column": 3,
                  "line": 3,
                  "offset": 18,
                },
              },
              "type": "paragraph",
            },
          ],
          "position": {
            "end": {
              "column": 16,
              "line": 3,
              "offset": 31,
            },
            "start": {
              "column": 1,
              "line": 3,
              "offset": 16,
            },
          },
          "spread": false,
          "type": "listItem",
        },
        {
          "checked": null,
          "children": [
            {
              "children": [
                {
                  "position": {
                    "end": {
                      "column": 21,
                      "line": 4,
                      "offset": 52,
                    },
                    "start": {
                      "column": 3,
                      "line": 4,
                      "offset": 34,
                    },
                  },
                  "type": "text",
                  "value": "1/2 lb ground beef",
                },
              ],
              "position": {
                "end": {
                  "column": 21,
                  "line": 4,
                  "offset": 52,
                },
                "start": {
                  "column": 3,
                  "line": 4,
                  "offset": 34,
                },
              },
              "type": "paragraph",
            },
          ],
          "position": {
            "end": {
              "column": 21,
              "line": 4,
              "offset": 52,
            },
            "start": {
              "column": 1,
              "line": 4,
              "offset": 32,
            },
          },
          "spread": false,
          "type": "listItem",
        },
      ],
      "ordered": false,
      "position": {
        "end": {
          "column": 21,
          "line": 4,
          "offset": 52,
        },
        "start": {
          "column": 1,
          "line": 3,
          "offset": 16,
        },
      },
      "spread": false,
      "start": null,
      "type": "list",
    },
  ],
  "position": {
    "end": {
      "column": 21,
      "line": 4,
      "offset": 52,
    },
    "start": {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
